# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/StructuredCloneHolder.h
# Commit: ea637545c497
# Full Hash: ea637545c4972c106a7e2ed801cb514b9412ff62
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-03-06 04:41:40
# Regressor Bug: 1858958
# File Overlap Count: 1
# Description:
#   Bug 1858958 - AudioData: structured cloning. r=chunmin,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D190948
# ==============================================================================

diff -r fdc20c3b80d6 -r ea637545c497 dom/base/StructuredCloneHolder.h
--- a/dom/base/StructuredCloneHolder.h	Tue Mar 05 15:52:12 2024 +0000
+++ b/dom/base/StructuredCloneHolder.h	Tue Mar 05 15:52:13 2024 +0000
@@ -169,6 +169,7 @@
 class MessagePort;
 class MessagePortIdentifier;
 struct VideoFrameSerializedData;
+struct AudioDataSerializedData;
 
 class StructuredCloneHolder : public StructuredCloneHolderBase {
  public:
@@ -270,6 +271,8 @@
 
   nsTArray<VideoFrameSerializedData>& VideoFrames() { return mVideoFrames; }
 
+  nsTArray<AudioDataSerializedData>& AudioData() { return mAudioData; }
+
   nsTArray<EncodedVideoChunkData>& EncodedVideoChunks() {
     return mEncodedVideoChunks;
   }
@@ -379,6 +382,9 @@
   // Used for cloning VideoFrame in the structured cloning algorithm.
   nsTArray<VideoFrameSerializedData> mVideoFrames;
 
+  // Used for cloning AudioData in the structured cloning algorithm.
+  nsTArray<AudioDataSerializedData> mAudioData;
+
   // Used for cloning EncodedVideoChunk in the structured cloning algorithm.
   nsTArray<EncodedVideoChunkData> mEncodedVideoChunks;
 